src/Umbraco.Infrastructure/PropertyEditors/BlockEditorValidatorBase.cs

```diff
@@ -16,16 +18,55 @@ protected BlockEditorValidatorBase(IPropertyValidationService propertyValidation
         : base(propertyValidationService)
         => _elementTypeCache = elementTypeCache;
 
-    protected IEnumerable<ElementTypeValidationModel> GetBlockEditorDataValidation(BlockEditorData<TValue, TLayout> blockEditorData)
+    protected IEnumerable<ElementTypeValidationModel> GetBlockEditorDataValidation(BlockEditorData<TValue, TLayout> blockEditorData, PropertyValidationContext validationContext)
+    {
+        var elementTypeValidation = new List<ElementTypeValidationModel>();
+        var isWildcardCulture = validationContext.Culture == "*";
+        var validationContextCulture = isWildcardCulture ? null : validationContext.Culture.NullOrWhiteSpaceAsNull();
+        elementTypeValidation.AddRange(GetBlockEditorDataValidation(blockEditorData, validationContextCulture, validationContext.Segment));
+
+        if (validationContextCulture is null)
+        {
+            IEnumerable<string> validationContextCulturesBeingValidated = isWildcardCulture
+                ? blockEditorData.BlockValue.Expose.Select(e => e.Culture).WhereNotNull().Distinct()
+                : validationContext.CulturesBeingValidated;
+            foreach (var culture in validationContextCulturesBeingValidated)
+            {
+                foreach (var segment in validationContext.SegmentsBeingValidated.DefaultIfEmpty(null))
+                {
+                    elementTypeValidation.AddRange(GetBlockEditorDataValidation(blockEditorData, culture, segment));
+                }
+            }
+        }
+
+        return elementTypeValidation;
+    }
+
+    private IEnumerable<ElementTypeValidationModel> GetBlockEditorDataValidation(BlockEditorData<TValue, TLayout> blockEditorData, string? culture, string? segment)
     {
         // There is no guarantee that the client will post data for every property defined in the Element Type but we still
         // need to validate that data for each property especially for things like 'required' data to work.
         // Lookup all element types for all content/settings and then we can populate any empty properties.
 
+        if (blockEditorData.Layout is null)
+        {
+            yield break;
+        }
+
+        Guid[] exposedContentKeys = blockEditorData.BlockValue.Expose
+            .Where(e => culture is null || e.Culture == culture)
+            .Select(e => e.ContentKey)
+            .Distinct()
+            .ToArray();
+        Guid[] exposedSettingsKeys = blockEditorData.Layout
+            .Where(l => l.SettingsKey.HasValue && exposedContentKeys.Contains(l.ContentKey))
+            .Select(l => l.SettingsKey!.Value)
+            .ToArray();
```

Comment:
IMO, if we want to use single letters we should use x, but I'd still rather be explicit 🙈 

```suggestion
        Guid[] exposedContentKeys = blockEditorData.BlockValue.Expose
            .Where(expose => culture is null || expose.Culture == culture)
            .Select(expose => expose.ContentKey)
            .Distinct()
            .ToArray();
        Guid[] exposedSettingsKeys = blockEditorData.Layout
            .Where(layout => layout.SettingsKey.HasValue && exposedContentKeys.Contains(layout.ContentKey))
            .Select(layout => layout.SettingsKey!.Value)
            .ToArray();
```